sctk/src/handlers/virtual_keyboard.rs

```diff
@@ -0,0 +1,123 @@
+use std::{fmt::Debug, marker::PhantomData};
+
+use iced_futures::core::event::wayland::KeyEvent;
+use sctk::reexports::client::{
+    delegate_dispatch,
+    globals::{BindError, GlobalList},
+    protocol::{wl_keyboard, wl_seat::WlSeat},
+    Connection, Dispatch, Proxy, QueueHandle,
+};
+
+use wayland_protocols_misc::zwp_virtual_keyboard_v1::client::{
+    zwp_virtual_keyboard_manager_v1::ZwpVirtualKeyboardManagerV1,
+    zwp_virtual_keyboard_v1::ZwpVirtualKeyboardV1,
+};
+
+use sctk::globals::GlobalData;
+
+use crate::event_loop::state::SctkState;
+
+use super::input_method::keyboard::RawModifiers;
+
+#[derive(Debug)]
+pub struct VirtualKeyboardManager<T> {
```

Comment:
I wonder if it would make sense to provide a simple helper for the virtual keyboard protocol in smithay-client-toolkit. Though it's not too complicated of a protocol.

Reply:
Virtual keyboard is really small and kind of niche, maybe it would make more sense a library/crate that can be used with sctk?

Reply:
Might be an idea to move a lot of the boilerplate code to its own crate, niche_sctk or something like that

Reply:
Personally I don't see an issue with having fairly niche things (that are commonly supported standard Wayland protocols) in sctk. As long as they don't significantly increase build time (then they could have feature flags) or distract too much from the more widely used parts of the crate. I don't know what others think about that though.

Reply:
I will make a pr for both and see what happens :)